tests/common/plugins/loganalyzer/loganalyzer.py

```diff
@@ -209,22 +214,31 @@ def analyze(self, marker, fail=True):
 
             # On DUT extract syslog files from /var/log/ and create one file by location - /tmp/syslog
             self.ansible_host.extract_log(directory='/var/log', file_prefix='syslog', start_string=start_string, target_filename=self.extracted_syslog)
+            if self.analyze_sairedis:
+                self.ansible_host.extract_log(directory='/var/log/swss', file_prefix='sairedis.rec', start_string='recording on: /var/log/swss/sairedis.rec', target_filename=self.extracted_sairedis_rec)
         finally:
             # Enable logrotate cron task back
             self.ansible_host.command("sed -i 's/^#//g' /etc/cron.d/logrotate")
 
         # Download extracted logs from the DUT to the temporal folder defined in SYSLOG_TMP_FOLDER
         self.save_extracted_log(dest=tmp_folder)
+        file_list = [tmp_folder]
+
+        if self.analyze_sairedis:
+            tmp_rec_folder = ".".join((SAIREDIS_TMP_FOLDER, timestamp))
+            self.save_extracted_sairedis_rec(dest=tmp_rec_folder)
+            file_list.append(tmp_rec_folder)
 
         match_messages_regex = re.compile('|'.join(self.match_regex)) if len(self.match_regex) else None
         ignore_messages_regex = re.compile('|'.join(self.ignore_regex)) if len(self.ignore_regex) else None
         expect_messages_regex = re.compile('|'.join(self.expect_regex)) if len(self.expect_regex) else None
 
-        analyzer_parse_result = self.ansible_loganalyzer.analyze_file_list([tmp_folder], match_messages_regex, ignore_messages_regex, expect_messages_regex)
+        analyzer_parse_result = self.ansible_loganalyzer.analyze_file_list(file_list, match_messages_regex, ignore_messages_regex, expect_messages_regex)
         # Print syslog file content and remove the file
-        with open(tmp_folder) as fo:
-            logging.debug("Syslog content:\n\n{}".format(fo.read()))
-        os.remove(tmp_folder)
+        for folder in file_list:
+            with open(folder) as fo:
+                logging.debug("Syslog content:\n\n{}".format(fo.read()))
```

Comment:
If analyze_sairedis is True, this should be `Sairedis content:`.

Reply:
Made the log message related to the file name. Also changed the logic to allow to specify files that need to be analyzed.